### PR TITLE
Use CI linters in pre-commit

### PR DESCRIPTION
**Pull request by amureki**
I noticed that we have pre-commit installed, also a bunch of linters running in CI.
However, I had no pre-commit running locally (wrong pre-commit config file), also there were no checks defined.

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,7 +0,0 @@
--   id: relint
```
Comment (codingjoe) [inline .pre-commit-hooks.yaml:1]: Why did you remove the hook?

File: linter-requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-bandit==1.7.4
+bandit[toml]==1.7.4
```
Comment (codingjoe) [inline linter-requirements.txt:1]: Cool change, but somewhat unrelated, right?

**Review by codingjoe — COMMENTED** [review]
I avoided setting up pre-commit on my projects in the past, mainly because of updates.
IMHO, this increases the barrier to entry for new contributors. I am very hesitant about this change.

File: linter-requirements.txt (hunk 2 above)
Comment (amureki) [reply linter-requirements.txt:1]: True, I just did it because I wanted to simplify bandit setup in the pre-commit.

**Comment by amureki** [comment]
@codingjoe I got your point, makes sense. I wonder if we could try https://pre-commit.ci/ (free for open-source repos), which will automatically fix all linting issues.
